codespell_lib/_codespell.py

```diff
@@ -228,7 +228,8 @@ def open_with_chardet(self, filename: str) -> Tuple[List[str], str]:
                     break
         self.encdetector.close()
         encoding = self.encdetector.result["encoding"]
-        assert encoding is not None
```

Comment:
I clearly don't know much about mypy, I don't understand why it would change the signature of the function if we remove `assert`, especially since assert could be [optimised away](https://docs.python.org/3/reference/simple_stmts.html#grammar-token-python-grammar-assert_stmt).

Reply:
Ah, got it. :blush:

Reply:
I would favour [locale.getencoding()](https://docs.python.org/3/library/locale.html#locale.getencoding) because it's the default value of the `encoding` argument of open(). Perhaps someone else could look into this... @larsoner?

Reply:
`locale.getencoding()` seems reasonable to me assuming it works. Or get mypy to skip its check or whatever is fine, too...

Reply:
Unfortunately [that method](https://docs.python.org/3.11/library/locale.html#locale.getencoding) is new in Python 3.11, so this will break with all lower Python versions.
How about we use [`getpreferredencoding(False)`](https://docs.python.org/3.11/library/locale.html#locale.getpreferredencoding) instead? This is what Python <= 3.10 use when `encoding` in `None` when calling `open`.

Reply:
Yes, doing whatever `open` does internally looks like the best course of action.

Reply:
Using `locale.getpreferredencoding(False)` now & rebased to fix merge conflict :v: